### PR TITLE
fix: Check for existence of_ SC_AVPHYS_PAGES to fix MacOS compilation. 

### PR DESCRIPTION
**Pull request by CusiniM**
The merge of [PR](https://github.com/GEOS-DEV/GEOS/pull/3052) has broken compilation on my mac laptop. I have patched it like this.

### CODE REVIEW

**Comment by rrsettgast** [comment]
Time to use the GitHub M1 runners?

**Review by rrsettgast — APPROVED** [review]
Is there another option on the Mac?

**Comment by CusiniM** [comment]
> Is there another option on the Mac?



> Is there another option on the Mac?

I found [this](https://stackoverflow.com/questions/30511579/linux-sc-phys-pages-not-working-on-mac-os-x) but it's kind of old and I have not tried it.

**Comment by rrsettgast** [comment]
> > Is there another option on the Mac?
> 
> > Is there another option on the Mac?
> 
> I found [this](https://stackoverflow.com/questions/30511579/linux-sc-phys-pages-not-working-on-mac-os-x) but it's kind of old and I have not tried it.

@arng40 Can you give it a shot? Do you have a Mac?

**Comment by arng40** [comment]
> > > Is there another option on the Mac?
> > 
> > 
> > > Is there another option on the Mac?
> > 
> > 
> > I found [this](https://stackoverflow.com/questions/30511579/linux-sc-phys-pages-not-working-on-mac-os-x) but it's kind of old and I have not tried it.
> 
> @arng40 Can you give it a shot? Do you have a Mac?

I don't have a MAC so it would be difficult to test alternatives.
A solution could be to integrate a compilation check on macos in CI

File: src/coreComponents/common/MemoryInfos.cpp [hunk 1]
```diff
@@ -27,8 +27,13 @@ MemoryInfos::MemoryInfos( umpire::MemoryResourceTraits::resource_type resourceTy
     case umpire::MemoryResourceTraits::resource_type::pinned:
       #if defined( _SC_PHYS_PAGES ) && defined( _SC_PAGESIZE )
       m_totalMemory = sysconf( _SC_PHYS_PAGES ) * sysconf( _SC_PAGESIZE );
+      #if defined(_SC_AVPHYS_PAGES)
       m_availableMemory = sysconf( _SC_AVPHYS_PAGES ) * sysconf( _SC_PAGESIZE );
       #else
+      GEOS_WARNING( "Unknown device avaialable memory size getter for this system." );
```
Comment (MelReyCG) [inline src/coreComponents/common/MemoryInfos.cpp:33]:
```suggestion
      GEOS_WARNING( "Unknown device available memory size getter for this system." );
```

File: src/coreComponents/common/MemoryInfos.cpp [hunk 2]
```diff
@@ -27,8 +27,13 @@ MemoryInfos::MemoryInfos( umpire::MemoryResourceTraits::resource_type resourceTy
     case umpire::MemoryResourceTraits::resource_type::pinned:
       #if defined( _SC_PHYS_PAGES ) && defined( _SC_PAGESIZE )
       m_totalMemory = sysconf( _SC_PHYS_PAGES ) * sysconf( _SC_PAGESIZE );
+      #if defined(_SC_AVPHYS_PAGES)
       m_availableMemory = sysconf( _SC_AVPHYS_PAGES ) * sysconf( _SC_PAGESIZE );
       #else
```
Comment (MelReyCG) [inline src/coreComponents/common/MemoryInfos.cpp:32]: Would something like this be a way to find the available memory on MaxOS? Unfortunately, I we don't have any MacOS to test that here.

```suggestion
      #elif /* is compiling for MacOS */
        {
          uint64_t mem;
          size_t len = sizeof( mem );
          sysctlbyname( "hw.memsize", &mem, &len, nullptr, 0 );
          m_availableMemory= mem / sysconf(_SC_PAGE_SIZE);
        }
```

Should we regroup the system-dependant component like this in a namespace ?

**Review by MelReyCG — COMMENTED** [review]
Hello there, a few comments to contribute on this issue.

**Comment by CusiniM** [comment]
> > > > Is there another option on the Mac?
> > > 
> > > 
> > > > Is there another option on the Mac?
> > > 
> > > 
> > > I found [this](https://stackoverflow.com/questions/30511579/linux-sc-phys-pages-not-working-on-mac-os-x) but it's kind of old and I have not tried it.
> > 
> > 
> > @arng40 Can you give it a shot? Do you have a Mac?
> 
> I don't have a MAC so it would be difficult to test alternatives. A solution could be to integrate a compilation check on macos in CI

is it now available? I thought that we could not find any M1 runner but if they now have one we really should. It's true that we don't really need to support MacOS but a lot of us work on macbooks so it would be convenient to have a CI job to cover it.

**Pull request merged**